Description: don't hack the libjpeg mem manager....
 When linking with a shared libjpeg library, don't include
 the non-public libjpeg API hack to replace it's low level memory
 manager with Ghostscript's. But still do so when using our "local"
 libjpeg source.
Origin: http://git.ghostscript.com/?p=ghostpdl.git;a=commit;h=ceef32
Author: Chris Liddell <chris.liddell@artifex.com>
Forwarded: yes
Bug: http://bugs.ghostscript.com/show_bug.cgi?id=692891
Bug-Debian: http://bugs.debian.org/653061
Last-Update: 2012-03-02

--- a/base/sjpegc.c
+++ b/base/sjpegc.c
@@ -35,6 +35,9 @@
  */
 
 #include "gconfig_.h"
+
+#if SHARE_JPEG == 0
+/* Don't use the non-public insterface if we're linking to a shared lib */
 #ifdef DONT_HAVE_JMEMSYS_H
 
 void *
@@ -67,6 +70,7 @@
 #else
 #include "jmemsys.h"		/* for prototypes */
 #endif
+#endif /* SHAREJPEG == 0 */
 
 private_st_jpeg_block();
 
@@ -173,6 +177,8 @@
     return 0;
 }
 
+#if SHARE_JPEG == 0
+/* Don't use the non-public insterface if we're linking to a shared lib */
 /*
  * These routines replace the low-level memory manager of the IJG library.
  * They pass malloc/free calls to the Ghostscript memory manager.
@@ -277,3 +283,4 @@
 {
     /* no work */
 }
+#endif /* SHAREJPEG == 0 */
